views/js/expressCheckout.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+/**
+ * Waiting that the dom is loaded
+ */
+document.addEventListener('DOMContentLoaded', () => {
+  // if express checkout is used on cart page (expressCheckout hook)
+  if (displayMode === 'cart') {
+    window.prestashop.on('updatedCart', () => {
+      initPaypalSmartButtons();
+    });
+  }
+
+  if (displayMode === 'product') {
+    const addToCartButton = document.getElementsByClassName('product-add-to-cart');
+    addToCartButton[0].parentNode.insertBefore(document.getElementById('pscheckout-express-checkout'), addToCartButton[0].nextSibling);
+  }
+
+  // wait paypal sdk to be fully loaded
+  const interval = setInterval(() => {
+    if (window.paypal !== undefined) {
+      initPaypalSmartButtons();
+      clearInterval(interval);
+    }
+  }, 200);
+});
+
+/**
+ * Init paypal buttons with the given paypal order
+ *
+ * @param {int} paypalOrder
+ */
+function initPaypalSmartButtons() {
+  const controllerLink = expressCheckoutController.replace(/\amp;/g, '');
+
+  window.paypal.Buttons({
+    style: {
+      size: 'responsive',
+      shape: 'pill',
+      label: 'pay',
+    },
+    onRender() {
+      cleanAllExpressCheckoutInstance();
+      showExpressCheckout();
+    },
+    createOrder(data, actions) {
+      return createPaypalOrder().then((result) => {
+        const paypalScript = document.getElementById('paypalSdkPsCheckout');
+        paypalScript.setAttribute('data-client-token', result.body.client_token);
+
+        return result.body.id;
+      }).catch(() => {
+        throw new Error('Not able to create a paypal order');
+      });
+    },
+    onApprove(paypal, actions) {
+      actions.order.get().then((order) => {
+        redirect(order, controllerLink);
+      });
+    },
+  }).render('#paypal-button-container');
+}
+
+/**
+ * Ajax request to create and retrieve paypal order
+ */
+async function createPaypalOrder() {
```

Comment:
To be tested on IE11. Looks like `async` isn't compatible with this browser https://caniuse.com/#feat=async-functions

Reply:
Compiling these JS scripts for compatibility with old browser may help